R/zzz.R

```diff
@@ -1,5 +1,27 @@
 .lockbox_env <- new.env()
 
+.get_safely_from_envir <- function(target, default = NULL) {
+  res <- tryCatch(
+    get(target, envir = .lockbox_env)
+  , error = function(e) {default})
+}
+
+set_session_id <- function() {
+  if (is.null(.get_safely_from_envir("session_id"))) {
+    if (requireNamespace("uuid", quietly = TRUE)) {
+      assign("session_id", uuid::UUIDgenerate(), envir = .lockbox_env)
```

Comment:
This bit is currently a bit of a mess because I'm not quite sure where to stick the finalizer / how we'll finalize is a bit in flux.